pkg/app/zipper/http_handlers.go

```diff
@@ -570,7 +569,7 @@ func (app *App) lbCheckHandler(w http.ResponseWriter, req *http.Request, ms *Pro
 func (app *App) filterBackendByTopLevelDomain(targets []string) []backend.Backend {
 	targetTlds := make([]string, 0, len(targets))
 	for _, target := range targets {
-		targetTlds = append(targetTlds, getTopLevelDomain(target))
+		targetTlds = append(targetTlds, getTargetTopLevelDomain(target, app.TLDPrefixes))
```

Comment:
nit: we are using both Tld and TopLevelDomain in same code, imho we could standardize on one of the two for consistency, maybe Tld since it's used also in package name and config_